moveit_planners/ompl/ompl_interface/src/model_based_planning_context.cpp

```diff
@@ -120,11 +124,49 @@ void ompl_interface::ModelBasedPlanningContext::configure(const ros::NodeHandle&
   if (spec_.constrained_state_space_)
   {
     // convert the input state to the corresponding OMPL state
-    ompl::base::ScopedState<> ompl_start_state(spec_.constrained_state_space_);
+    ob::ScopedState<> ompl_start_state(spec_.constrained_state_space_);
     spec_.state_space_->copyToOMPLState(ompl_start_state.get(), getCompleteInitialRobotState());
     ompl_simple_setup_->setStartState(ompl_start_state);
     ompl_simple_setup_->setStateValidityChecker(
         ob::StateValidityCheckerPtr(std::make_shared<ConstrainedPlanningStateValidityChecker>(this)));
+    ROS_INFO_STREAM("start state from getCompleteInitialRobotState: " << getCompleteInitialRobotState());
+
+    std::map<std::string, std::string> cfg = getSpecificationConfig();
+    auto it = cfg.find("constrained_state_space");
+    if (it != cfg.end())
+    {
+      // extract joint state goal from planning request
+      // auto constraints = request_.goal_constraints[0];
+      // ROS_INFO_STREAM("goal constraints from planning request: " << constraints);
+
+      // ob::ScopedState<> goal(spec_.constrained_state_space_);
+      // goal->as<ob::ConstrainedStateSpace::StateType>()->copy(goal_joint_positions);
+      // ROS_INFO_STREAM("goal reconstructed from planning request: " << goal);
+
+      // for Atlas and TangentBundle, the start and goal states have to be anchored.
+      if (it->second == "TangentBundleStateSpace")
+      {
+        ROS_INFO_NAMED("model_based_planning_context",
+                       "trying to anchor start states for constrained tangent bundle state space.");
+        spec_.constrained_state_space_->as<ob::TangentBundleStateSpace>()->anchorChart(ompl_start_state.get());
+        ROS_INFO_NAMED("model_based_planning_context",
+                       "trying to anchor goal states for constrained tangent bundle state space.");
+        // spec_.constrained_state_space_->as<ob::TangentBundleStateSpace>()->anchorChart(goal.get());
```

Comment:
We need the goal as a joint state here to be able to ``anchorChart()``.
Or maybe anchorChart should be called somewhere else in the interface?